File: dotnet/src/Cekat.EventSdk/EventInput.cs
namespace Cekat.EventSdk;

/// <summary>
/// An identity-bearing event. At least one of <see cref="Email"/> or <see cref="PhoneNumber"/> must be nonblank.
/// Identity strings are sent unchanged; they are trimmed only to check for blank values.
/// </summary>
/// <param name="Email">The contact email address.</param>
/// <param name="PhoneNumber">The contact phone number.</param>
/// <param name="ContactName">The contact display name.</param>
/// <param name="Properties">
/// A string-keyed object: an <c>IDictionary</c> with string keys, an
/// <c>IEnumerable&lt;KeyValuePair&lt;string, object?&gt;&gt;</c>, a <c>JsonObject</c>, or an object <c>JsonElement</c>.
/// Values may be null, booleans, strings, finite numbers within ±9,007,199,254,740,991 when integral, lists, and nested objects.
/// </param>
/// <param name="VisitorId">An explicit visitor ID. When blank, the current visitor context is used.</param>
/// <param name="EventId">A deduplication ID. When blank, the SDK generates a random UUID reused by every retry.</param>
/// <param name="OccurredAt">When the event happened. Defaults to the time of the call.</param>
public sealed record EventInput(
    string? Email = null,
    string? PhoneNumber = null,
    string? ContactName = null,
    object? Properties = null,
    string? VisitorId = null,
    string? EventId = null,
    DateTimeOffset? OccurredAt = null);

File: ruby/lib/cekat_event_sdk/event_input.rb
# frozen_string_literal: true

module CekatEventSdk
  # An identity-bearing event. At least one of email or phone_number must be nonblank.
  #
  # properties is nil (omitted) or a Hash with String or Symbol keys. Values may be nil, true,
  # false, Strings, Symbols, finite numbers (Integer, Float, BigDecimal, Rational), Arrays, and
  # Hashes of those. event_id deduplicates retried deliveries and defaults to a random UUID;
  # occurred_at (a Time or DateTime) defaults to the time of the call.
  EventInput = Data.define(:email, :phone_number, :contact_name, :visitor_id, :properties, :event_id, :occurred_at) do
    def initialize(email: nil, phone_number: nil, contact_name: nil, visitor_id: nil, properties: nil, event_id: nil, occurred_at: nil)
      super
    end
  end
end
